### PR TITLE
virtualisation: Use system.build.image, normalize file names

### PR DESCRIPTION
**Pull request by phaer**
This uses changes in https://github.com/NixOS/nixpkgs/pull/359326 and https://github.com/NixOS/nixpkgs/pull/359328 to normalize file names of disk images generated by modules in virtualisation. Please see the PR linked below for a table listing all changed filenames

It was split off https://github.com/NixOS/nixpkgs/pull/347275 in order to upstream that one in smaller chunks, easier to review, test and (hopefully not) revert.

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: nixos/modules/image/images.nix [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  lib,
+  pkgs,
+  extendModules,
+  ...
+}:
+let
+  inherit (lib) types;
+
+  imageModules = {
+    azure = [ ../virtualisation/azure-image.nix ];
```
[the remark this answers is not in the document]
Comment (phaer) [reply nixos/modules/image/images.nix:12]: Its in https://github.com/NixOS/nixpkgs/pull/359345. While splitting the PR, i decided to do the modules in 2 PRs. One PR per module seemed overkill with ~20 modules.

**Review by zimbatm — APPROVED** [review]

**Review by adamcstephens — APPROVED** [review]
Reviewing this for the lxc container changes, I think it looks good. Thanks for undertaking this standardization. :heart:

**Comment by zimbatm** [comment]
Let's merge this before it diverges?

**Comment by phaer** [comment]
Just rebased, if it still looks good it might be good to merge. I did *not* test a full rebuild of all image since the rebase yet though and probably won't get to it before Monday.

**Comment by zimbatm** [comment]
Passed nixpkgs-review :+1:

**Pull request merged**